### PR TITLE
Bringup Featured Circuits

### PR DESCRIPTION
**Pull request by manjotsidhu**
With the attempt to increase discovery of Circuits #30, this Pull Request adds Featured Circuits View and also shows top 3 featured circuits in Home View.

1. Adds Featured Circuits View
2. Adds Explore (Expandable Tile) > Featured Circuits in App Drawer.
3. Adds Editor Picks in Home which shows top 3 featured circuits.

Preview -

ezgif-4-a0ff42f3f355

Screenshots -

### CODE REVIEW

File: lib/ui/views/projects/featured_projects_view.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+          );
+        }
+
+        _items.insert(
+            0,
+            CVHeader(
+              title: 'Editor Picks',
+              description:
+                  'These circuits have been hand-picked by our authors for their awesomeness.',
+            ));
```
Comment (Nitish145) [inline lib/ui/views/projects/featured_projects_view.dart:68]: Better formatting can be done here

**Review by Nitish145 — APPROVED** [review]

**Comment by Nitish145** [comment]
Amazing work !!

File: lib/ui/views/projects/featured_projects_view.dart (hunk 1 above)
Comment (manjotsidhu) [reply lib/ui/views/projects/featured_projects_view.dart:68]: Sure will do it.

**Comment by manjotsidhu** [comment]
UI Test for Featured Projects has been added. Also, We need to keep in mind, Featured Projects and its details should can accessed even without the need to login.

* LocalStorageService -> `_getFromDisk` returns "null" (String) if user signs out.
* Project > `hasAuthorAccess` throws error if current user is signed out or never signed in.
* Profile > Fix viewing profile of other people without signing in.
* Project Details > Fix viewing the details of project without signing in.

These bugs have been fixed, now a new user can directly check public project details, profiles without even needing to sign in.

**Comment by manjotsidhu** [comment]
@Nitish145 @tachyons please review

**Pull request merged**